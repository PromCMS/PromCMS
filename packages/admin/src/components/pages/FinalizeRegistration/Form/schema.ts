import { object, string, ref } from 'yup';

export const finalizeRegistrationFormSchema = object()
  .shape({
    new_password: string().required('Dont forget to include this value'),
    confirmed_new_password: string()
      .required('Dont forget to include this value')
      .oneOf([ref('new_password'), null], 'Passwords must match'),
    token: string().required('Dont forget to include this value'),
  })
  .noUnknown();
