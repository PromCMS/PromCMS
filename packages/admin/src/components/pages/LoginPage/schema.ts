import { MESSAGES } from '@constants'
import { number, object, string } from 'yup'

export const loginFormSchema = object()
  .shape({
    step: number().required(),
    password: string().when('step', {
      is: 0,
      then: string().required(MESSAGES.PLEASE_ENTER_PASSWORD),
    }),
    email: string()
      .email()
      .when('step', {
        is: 0,
        then: string().required(MESSAGES.PLEASE_ENTER_EMAIL),
      }),
    token: string().when('step', {
      is: 2,
      then: string()
        .required(MESSAGES.PLEASE_ENTER_MFA_TOKEN)
        .length(6, MESSAGES.MFA_TOKEN_SHORT),
    }),
  })
  .noUnknown()
