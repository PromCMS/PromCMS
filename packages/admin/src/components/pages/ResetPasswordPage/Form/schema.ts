import { object, string, ref } from 'yup'

export const initializeResetPasswordFormSchema = object()
  .shape({
    email: string()
      .email('Please enter valid email address')
      .required('Dont forget to include this value'),
  })
  .noUnknown()

export const finalizeResetPasswordFormSchema = object()
  .shape({
    new_password: string().required('Dont forget to include this value'),
    confirmed_new_password: string()
      .required('Dont forget to include this value')
      .oneOf([ref('new_password'), null], 'Passwords must match'),
    token: string().required('Dont forget to include this value'),
  })
  .noUnknown()
