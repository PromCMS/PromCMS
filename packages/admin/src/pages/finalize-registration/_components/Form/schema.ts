import { object, string, ref } from 'yup';

export const finalizeRegistrationFormSchema = object()
  .shape({
    new_password: string().required('This field is required'),
    confirmed_new_password: string()
      .required('This field is required')
      .oneOf([ref('new_password'), null], 'Passwords must match'),
    token: string().required('This field is required'),
  })
  .noUnknown();
