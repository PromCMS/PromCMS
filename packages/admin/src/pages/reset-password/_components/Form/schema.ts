import { MESSAGES } from '@constants';
import { z } from 'zod';

export const initializeResetPasswordFormSchema = z
  .object({
    email: z
      .string({
        required_error: MESSAGES.FIELD_REQUIRED,
      })
      .email('Please enter valid email address'),
  })
  .strict();

export const finalizeResetPasswordFormSchema = z
  .object({
    new_password: z.string({
      required_error: MESSAGES.FIELD_REQUIRED,
    }),
    confirmed_new_password: z.string({
      required_error: MESSAGES.FIELD_REQUIRED,
    }),
    token: z.string({
      required_error: MESSAGES.FIELD_REQUIRED,
    }),
  })
  .strict()
  .superRefine((value, ctx) => {
    if (value.new_password !== value.confirmed_new_password) {
      ctx.addIssue({
        code: z.ZodIssueCode.custom,
        message: MESSAGES.PASSWORDS_MUST_MATCH,
      });
    }
  });
