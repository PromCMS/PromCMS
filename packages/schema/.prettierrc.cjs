module.exports = require('@prom-cms/config/default.prettier.cjs');
