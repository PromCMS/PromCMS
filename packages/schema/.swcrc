{
  "$schema": "http://json.schemastore.org/swcrc",
  "module": {
    "type": "es6"
  },
  "jsc": {
    "target": "esnext",
    "baseUrl": "./src",
    "parser": {
      "syntax": "typescript",
      "tsx": true
    },
    "transform": {
      "react": {
        "runtime": "automatic"
      }
    }
  }
}