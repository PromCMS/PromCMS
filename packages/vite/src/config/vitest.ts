import reactPlugin from '@vitejs/plugin-react';
import tsconfigPaths from 'vite-tsconfig-paths';
import { defineConfig } from 'vitest/config';

export default defineConfig({
  plugins: [
    tsconfigPaths(),
    // @ts-expect-error
    reactPlugin(),
  ],
  test: {},
});
