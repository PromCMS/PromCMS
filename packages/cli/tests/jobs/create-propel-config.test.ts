import path from 'path';
import fs from 'fs-extra';
import { describe, beforeEach, it, afterAll, expect } from 'vitest';
import createPropelConfig from '../../src/jobs/create-propel-config.js';

import { z } from 'zod';
import { MONOREPO_ROOT } from '@constants';
import { databaseConfigSchema } from '@prom-cms/schema';

const TEST_FOLDER_PATH = path.join(
  MONOREPO_ROOT,
  'node_modules',
  '.prom-cms',
  'test-app'
);
describe('jobs', () => {
  beforeEach(async () => {
    await fs.remove(TEST_FOLDER_PATH);
    await fs.ensureDir(TEST_FOLDER_PATH);
  });

  afterAll(async () => {
    await fs.remove(TEST_FOLDER_PATH);
  });

  describe('createPropelConfig', () => {
    it('should run correctly and generate files', async () => {
      const root = path.join(TEST_FOLDER_PATH, 'modules', 'test');
      await fs.ensureDir(root);

      await createPropelConfig({
        config: {
          project: { name: 'Testing Project', url: '' },
          database: databaseConfigSchema.parse({
            connections: [
              {
                adapter: 'sqlite',
                dsn: `sqlite:${path.join(
                  TEST_FOLDER_PATH,
                  '.database',
                  'application.sq3'
                )}`,
                name: 'default-connection',
                password: 'root',
                user: 'root',
              },
            ],
            models: [
              {
                tableName: 'first',
                admin: { hidden: true, icon: 'Article' },
                columns: [
                  { name: 'col', type: 'string', title: 'New' },
                  {
                    name: 'bola',
                    type: 'boolean',
                    localized: true,
                    title: 'Boolean Column',
                    default: false,
                  },
                ],
              },
              {
                admin: { icon: 'Article' },
                tableName: 'second_table_name',
                api: { enabled: false },
                columns: [{ name: 'col', type: 'string', title: 'Newo' }],
              },
            ],
          } satisfies z.input<typeof databaseConfigSchema>),
        },
        appRoot: TEST_FOLDER_PATH,
      });

      const expectedFiles = [
        path.join(TEST_FOLDER_PATH, '.prom-cms/propel/propel.json'),
        path.join(TEST_FOLDER_PATH, '.prom-cms/propel/schema.xml'),
      ];

      for (const filePath of expectedFiles) {
        expect(
          await fs.readFile(filePath, {
            encoding: 'utf8',
          })
        ).toMatchSnapshot();
      }
    });
  });
});
