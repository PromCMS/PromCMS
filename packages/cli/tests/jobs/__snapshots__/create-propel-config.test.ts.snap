// Vitest Snapshot v1

exports[`jobs > createPropelConfig > should run correctly and generate files 1`] = `
"{
  \\"propel\\": {
    \\"paths\\": {
      \\"schemaDir\\": \\"./.prom-cms/propel\\",
      \\"phpDir\\": \\"./modules/Testing/Models\\",
      \\"phpConfDir\\": \\"./.prom-cms/propel/config\\",
      \\"migrationDir\\": \\"./.prom-cms/propel/migrations\\",
      \\"sqlDir\\": \\"./.prom-cms/propel/sql\\"
    },
    \\"generator\\": {
      \\"namespaceAutoPackage\\": false
    },
    \\"database\\": {
      \\"connections\\": {
        \\"default-connection\\": {
          \\"adapter\\": \\"sqlite\\",
          \\"dsn\\": \\"sqlite:/Users/daichi/Desktop/Work/prom-cms/prom/node_modules/.prom-cms/test-app/.database/application.sq3\\",
          \\"user\\": \\"root\\",
          \\"password\\": \\"root\\",
          \\"settings\\": {
            \\"charset\\": \\"utf8\\"
          }
        }
      }
    }
  }
}
"
`;

exports[`jobs > createPropelConfig > should run correctly and generate files 2`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>
<database
  name=\\"default-connection\\"
  defaultIdMethod=\\"native\\"
  namespace=\\"PromCMS\\\\Modules\\\\Testing\\\\Models\\"
>
    <table
    name=\\"first\\"
    phpName=\\"First\\"
    prom.ignoreSeeding=\\"false\\"
    prom.title=\\"\\"
    prom.preset=\\"\\"
    prom.adminMetadata.icon=\\"Article\\"
    prom.adminMetadata.hidden=\\"true\\"
  >
            <column
      name=\\"col\\"
      type=\\"VARCHAR\\"
      required=\\"true\\"
      prom.hide=\\"false\\"
      prom.title=\\"New\\"
      prom.type=\\"string\\"
      prom.readonly=\\"false\\"
      prom.adminMetadata.isHidden=\\"false\\"
      prom.adminMetadata.editor.placement=\\"main\\"
      prom.adminMetadata.editor.width=\\"12\\"
      primaryKey=\\"false\\"
    />
            <column
      name=\\"bola\\"
      type=\\"BOOLEAN\\"
      required=\\"true\\"
      prom.hide=\\"false\\"
      prom.title=\\"Boolean Column\\"
      prom.type=\\"boolean\\"
      prom.readonly=\\"false\\"
      prom.adminMetadata.isHidden=\\"false\\"
      prom.adminMetadata.editor.placement=\\"main\\"
      prom.adminMetadata.editor.width=\\"12\\"
      defaultValue=\\"false\\"
      primaryKey=\\"false\\"
    />
        <behavior name=\\"i18n\\">
          <parameter name=\\"i18n_columns\\" value=\\"bola\\" />
        </behavior>
      <behavior name=\\"prom_model\\" />
    </table>
    <table
    name=\\"second_table_name\\"
    phpName=\\"SecondTableName\\"
    prom.ignoreSeeding=\\"false\\"
    prom.title=\\"\\"
    prom.preset=\\"\\"
    prom.adminMetadata.icon=\\"Article\\"
    prom.adminMetadata.hidden=\\"false\\"
  >
            <column
      name=\\"col\\"
      type=\\"VARCHAR\\"
      required=\\"true\\"
      prom.hide=\\"false\\"
      prom.title=\\"Newo\\"
      prom.type=\\"string\\"
      prom.readonly=\\"false\\"
      prom.adminMetadata.isHidden=\\"false\\"
      prom.adminMetadata.editor.placement=\\"main\\"
      prom.adminMetadata.editor.width=\\"12\\"
      primaryKey=\\"false\\"
    />
      <behavior name=\\"prom_model\\" />
    </table>
</database>
"
`;
